git-hooks/pre-push

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env php
+
+<?php
+
+$projectName = basename(getcwd());
+
+exec('vendor/bin/phpunit', $output, $returnCode);
+
+if ($returnCode !== 0) {
+  $minimalTestSummary = array_pop($output);
+  printf("Test suite for %s failed: ", $projectName);
+  printf("( %s ) %s%2\$s", $minimalTestSummary, PHP_EOL);
```

Comment:
https://secure.php.net/manual/en/function.sprintf.php
```
$num = 5;
$location = "tree"
$format = 'The %2$s contains %1$d monkeys';
echo sprintf($format, $num, $location);
```
Its just fancy printf string formatting.  It prints how "x of y test cases failed".

You're right though this code is confusing.  I'll simplify it.